oracle/blockoracle.go

```diff
@@ -124,8 +130,30 @@ func (bo *MinerBlockOracle) calcEligibilityProofs(epochNumber types.EpochID) err
 	}
 	bo.proofsEpoch = epochNumber
 	bo.eligibilityMutex.RLock()
-	bo.log.Info("miner %v is eligible for %d blocks on %d layers in epoch %d",
-		bo.nodeID.Key[:5], numberOfEligibleBlocks, len(bo.eligibilityProofs), epochNumber)
+
+	// Sort the layer map so we can print the layer data in order
+	keys := []types.LayerID{}
+	for k := range bo.eligibilityProofs {
+		if len(bo.eligibilityProofs[k]) > 0 {
+			keys = append(keys, k)
+		}
+	}
+	sort.Slice(keys, func(i, j int) bool {
+		return uint64(keys[i]) < uint64(keys[j])
+	})
+
+	// Pretty-print the number of blocks per eligible layer
+	strs := []string{}
+	for k := range keys {
+		strs = append(strs, fmt.Sprintf("Layer %d: %d", uint64(k), keys[k]))
+	}
```

Comment:
You've got this wrong...

The `k` here is the index of the layer ID in the slice (an enumeration), while `keys[k]` is the actual layer ID.

```suggestion
	for k := range keys {
		strs = append(strs, fmt.Sprintf("Layer %d: %d", keys[k], len(bo.eligibilityProofs[k])))
	}
```